internal/gatewayapi/translator.go

```diff
@@ -777,6 +899,71 @@ func isValidCrossNamespaceRef(from crossNamespaceFrom, to crossNamespaceTo, refe
 	return false
 }
 
+// Checks if a hostname is valid according to RFC 1123 and gateway API's requirement that it not be an IP address
+func isValidHostname(hostname string) (valid bool, reason string) {
```

Comment:
Any reason not to use [`k8s.io/apimachinery/pkg/util/validation.IsDNS1123Subdomain`](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/validation#IsDNS1123Subdomain)?

Reply:
I sorta hate that this returns a `string` instead of an `error`.

Reply:
See my response to your other comment re: using `IsDNS1123Subdomain`

Reply:
Actually I've switched to using `IsDNS1123Subdomain` along with the check for label length (and a test case for the last label as well)